src/support/progress_manager.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <mutex>
+#include <atomic>
+
+class ProgressManager
```

Comment:
With multithreading, we can no longer just set progress linearly in all elements. We instead have to have a centralized processor of progress.
In the future, this could be expanded to give more detailed/accurate progress, but for now, it's just enough for multi threading support.

Reply:
It may be worth moving this to common items in the future.

@Zemurin FYI